ansible/roles/test/tasks/bgp_multipath_relax.yml

```diff
@@ -0,0 +1,93 @@
+##########################################################
+### playbook to test leaf router BGP multipath relex
+########################################################## 
+
+- fail: msg="please provide testbed_type for bgp_multipath_relax test"
+  when: testbed_type is not defined 
+
+- fail: mgs="This test only works for leaf routers as DUT in topology, t1, t1-lag, t1-lag-64"
+  when: testbed_type not in ['t1', 't1-lag', 't1-64-lag']
+
+- name: Gathering minigraph facts about the device
+  minigraph_facts: host={{ inventory_hostname }}
+  connection: local
+
+- name:  Find all V4 bgp neighbors from minigraph
+  set_fact:
+    bgp_v4nei: "{{ bgp_v4nei | default({}) | combine({ item['name']: item['addr'] }) }}"
+  with_items: minigraph_bgp
+  when: "'::' not in item['addr']"
+
+- include_vars: "vars/topo_{{ testbed_type }}.yml"
+
+- name: get all t0 routers names
+  set_fact:
+    dut_t0_nei: "{{ dut_t0_nei|default([]) + [ item ] }}"
+  with_items: "{{ topology['VMs'] }}"
+  when: '"T0" in item'
+
+- name: get all t2 routers names
+  set_fact:
+    dut_t2_nei: "{{ dut_t2_nei|default([]) + [ item ] }}"
+  with_items: "{{ topology['VMs'] }}"
+  when: '"T2" in item'
+
+- name: find all t0 routers with vips defined and vips information
+  set_fact: 
+     vips_t0: "{{ vips_t0 | default([]) + [item] }}"
+     t0_asns: "{{ t0_asns | default([]) + [configuration[item]['bgp']['asn']] }}"
+     vips_prefixes: "{{ configuration[item]['vips']['ipv4']['range'] }}"
+     vips_asn: "{{ configuration[item]['vips']['ipv4']['asn'] }}"
+  with_items: "{{ dut_t0_nei }}"
+  when:  "{{ 'vips' in configuration[item].keys() }}" 
```

Comment:
this seems little bit complicated, and it is not guarantee to work if not all t0 advertise same vip prefixes. The assumption here is that all t0 advertising same set of vip prefixes.

I would suggest to add vip section (duplicated) in configuration_properties

- vips:
     prefix: 
       - 20.0.1.0/26
     t0: 
       - arista01t0
       - arista01t1

it is duplicated data, but the ansible logic is easier.

Reply:
yeah, this would  only work when all vips advertise same set of prefixes. change the topology yaml file like suggested here, will need carefully synced with configuration with each VM and the topology for whole testbed.
I changed the playbook to pick a prefix to validate that prefix's path.